File: benchmark/results/python-invocation-script/run-001/b-baseline/output/metaengine_demo/identity/services/user_repository.py
"""UserRepository service."""
from typing import Optional

from metaengine_demo.identity.aggregates.user import User


class UserRepository:
    """UserRepository service."""

    def create(self, input: Optional[User]) -> User:
        raise NotImplementedError("not implemented")

    def find_by_id(self, id: str) -> Optional[User]:
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[User]:
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-script-sonnet/run-001/b-baseline/output/src/domain/identity/services/UserRepository.ts
import { User } from '../aggregates/User';

/** UserRepository service. */
export class UserRepository {
  create(input: Partial<User>): User { throw new Error('not implemented'); }
  findById(id: string): User | null { throw new Error('not implemented'); }
  list(limit: number): User[] { throw new Error('not implemented'); }
  delete(id: string): void { throw new Error('not implemented'); }
}
